tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/async/JaxRsServerAsyncTest.java

```diff
@@ -131,6 +144,61 @@ private void doAsyncTest(Function<JaxRsServerAsyncTestEndpointClient, String> re
         // Assert baggage propagated on subtask span
         Assert.assertTrue(subtaskSpan.getAttributes().get(BAGGAGE_VALUE_ATTR).contains(TEST_VALUE));
 
+        // Assert that query parameter was passed correctly
+        Assert.assertTrue(serverSpan.getAttributes().get(HTTP_TARGET).contains(QUERY_VALUE));
+
+        // Assert that the server span finished after the subtask span
+        // Even though the resource method returned quickly, the span should not end until the response is actually
+        // returned
+        Assert.assertTrue(serverSpan.getEndEpochNanos() >= subtaskSpan.getEndEpochNanos());
+    }
+
+    private void doErrorAsyncTest(Function<JaxRsServerAsyncTestEndpointClient, String> requestFunction) {
+        Baggage baggage = Baggage.builder()
+                .put(JaxRsServerAsyncTestEndpoint.BAGGAGE_KEY, TEST_VALUE)
+                .build();
+
+        try (Scope s = baggage.makeCurrent()) {
+            // Make the request to the test endpoint
+            try {
+                JaxRsServerAsyncTestEndpointClient client = RestClientBuilder.newBuilder()
+                        .baseUri(url.toURI())
+                        .build(JaxRsServerAsyncTestEndpointClient.class);
+                try {
+                    requestFunction.apply(client);
+                } catch (WebApplicationException e) {
+                    readErrorSpans();
+                }
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void readErrorSpans() {
+        List<SpanData> spanData = spanExporter.getFinishedSpanItems(3);
+
+        SpanData subtaskSpan = spanExporter.getFirst(SpanKind.INTERNAL);
+        SpanData clientSpan = spanExporter.getFirst(SpanKind.CLIENT);
+        SpanData serverSpan = spanExporter.getFirst(SpanKind.SERVER);
+
+        // Assert correct parent-child links
+        // Shows that propagation occurred
+        assertEquals(serverSpan.getSpanId(), subtaskSpan.getParentSpanId());
+        assertEquals(clientSpan.getSpanId(), serverSpan.getParentSpanId());
+
+        // Assert error is in span from method subtaskError()
+        assertEquals(HTTP_INTERNAL_ERROR, clientSpan.getAttributes().get(HTTP_STATUS_CODE).intValue());
```

Comment:
The comment says we're asserting something about subtaskError, but the code is checking the client span.

It's also important that we check the response code from the server span.